packages/gatsby-mdx/mdx-renderer.js

```diff
@@ -1,27 +1,34 @@
 const React = require("react");
-const { withMDXComponents } = require("@mdx-js/react");
 const mdx = require("@mdx-js/react/create-element");
-const { withMDXScope } = require("./context");
+const { useMDXComponents } = require("@mdx-js/react");
+const { useMDXScope } = require("./context");
 
-module.exports = withMDXScope(
-  withMDXComponents(({ scope = {}, components = {}, children, ...props }) => {
-    if (!children) {
-      return null;
-    }
-    const fullScope = {
-      // React is here just in case the user doesn't pass them in
-      // in a manual usage of the renderer
-      React,
-      mdx,
-      ...scope
-    };
+module.exports = function MDXRenderer({
+  scope = {},
+  components = {},
+  children,
+  ...props
```

Comment:
I wonder if we need to transpile this file with babel since we use rest syntax? 🤔 
Related #327

Reply:
We compile ourselves with the babel config of the user: https://github.com/ChristopherBiscardi/gatsby-mdx/blob/c65afbc05a11f441739cc3827686452fddc109a2/packages/gatsby-mdx/gatsby/create-webpack-config.js#L21-L25